expression/builtin_cast.go

```diff
@@ -1440,7 +1440,11 @@ func (b *builtinCastStringAsDecimalSig) evalDecimal(row chunk.Row) (res *types.M
 	res = new(types.MyDecimal)
 	sc := b.ctx.GetSessionVars().StmtCtx
 	if !(b.inUnion && mysql.HasUnsignedFlag(b.tp.GetFlag()) && isNegative) {
-		err = sc.HandleTruncate(res.FromString([]byte(val)))
+		err = res.FromString([]byte(val))
```

Comment:
```suggestion
		err = sc.HandleTruncate(res.FromString([]byte(val)))
```
Maybe this is better? We handled `TruncateAsWarning` and `IgnoreTruncate` in `HandleTruncate` function.